irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -96,37 +55,52 @@ void OnDemandOrderingGate::propagateBatch(
       transport::OdOsNotification::CollectionType{batch});
 }
 
-rxcpp::observable<network::OrderingEvent> OnDemandOrderingGate::onProposal() {
-  return proposal_notifier_.get_observable();
+void OnDemandOrderingGate::processRoundSwitch(RoundSwitch const &event) {
+  log_->debug("Current: {}", event.next_round);
+  current_round_ = event.next_round;
+  current_ledger_state_ = event.ledger_state;
+
+  std::shared_lock<std::shared_timed_mutex> stop_lock(stop_mutex_);
+  if (stop_requested_) {
+    log_->warn("Not doing anything because stop was requested.");
+    return;
+  }
```

Comment:
Very strange defence

Reply:
Well, it was there. We can make a task to refactor it, but I think not in this PR.